ci/checks/ensure-release-notes-up-to-date.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+current_version=$(grep '__version__ = ' git_machete/__init__.py | cut -d\' -f2)
+release_notes_version=$(egrep -o '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?' RELEASE_NOTES.md --max-count 1)
```

Comment:
I managed to find official regex for semantic version matching, at the very end of site: https://semver.org/, should we use this or the previous, simpler one ( `\d{1,}.\d{1,}.\d{1,}`)?

Reply:
![image](https://user-images.githubusercontent.com/3383210/151787215-8bbeac68-7445-4206-8ebb-0cb0415c1f84.png)

Reply:
Let's stick to the short one! I don't plan on ever using build metadata/pre-release versions

Reply:
![image](https://user-images.githubusercontent.com/86779115/151845440-0f2dfdea-0472-4f49-b0e5-dccb6b210603.png)